doc/src/guides/solving/solve-ode.md

```diff
@@ -0,0 +1,393 @@
+you can extract results by specifying the index of each function ("slicing" each
+solution), we recommend an approach which is robust with respect to function
+ordering. The following converts each solution into a dictionary so you can
+easily extract the result for the desired function. It uses standard Python
+techniques such as a loops or comprehensions, in a nested fashion.
+
+```py
+>>> from sympy import symbols, Eq, Function, dsolve
+>>> y, z = symbols("y z", cls=Function)
+>>> x = symbols("x")
+>>> eqs = [Eq(y(x).diff(x)**2, z(x)**2), Eq(z(x).diff(x), z(x))]
+>>> solutions = dsolve(eqs, [y(x), z(x)])
+>>> solutions
+[[Eq(y(x), C1 - C2*exp(x)), Eq(z(x), C2*exp(x))], [Eq(y(x), C1 + C2*exp(x)), Eq(z(x), C2*exp(x))]]
+>>> solutions_list = [] # nested list approach
+>>> for solution in solutions:
+...     solution_dict = {}
+...     for fn in solution:
+...             solution_dict.update({fn.lhs: fn.rhs})
+...     solutions_list.append(solution_dict)
+>>> solutions_list
+[{y(x): C1 - C2*exp(x), z(x): C2*exp(x)}, {y(x): C1 + C2*exp(x), z(x): C2*exp(x)}]
+>>> solutions_list = [{fn.lhs:fn.rhs for fn in solution} for solution in solutions]
+>>> solutions_list # nested comprehension approach
+[{y(x): C1 - C2*exp(x), z(x): C2*exp(x)}, {y(x): C1 + C2*exp(x), z(x): C2*exp(x)}]
+>>> solutions_list[0][y(x)]
+C1 - C2*exp(x)
+```
+
+### Work With Arbitrary Constants
+
+You can manipulate arbitrary constants such as `C1`, `C2`, and `C3`, which are
+generated automatically by {func}`~.dsolve`, by creating them as symbols. For
+example, if you want to assign values to arbitrary constants, you can create
+them as symbols and then substitute in their values using
+{meth}`~sympy.core.basic.Basic.subs`:
+
+```py
+>>> from sympy import Function, dsolve, Derivative, symbols, pi
+>>> x, C1, C2 = symbols("x, C1, C2")
+>>> y = Function('y')
+>>> result = dsolve(Derivative(y(x), x, x) + 9*y(x), y(x)).rhs
+>>> result
+C1*sin(3*x) + C2*cos(3*x)
+>>> result.subs({C1: 7, C2: pi})
+7*sin(3*x) + pi*cos(3*x)
+```
+
+### Numerically Solve an ODE in SciPy
+
+A common workflow which leverages
+[SciPy's](https://docs.scipy.org/doc/scipy/index.html) fast numerical ODE
+solving is
+1. set up an ODE in SymPy
+2. convert it to a lambda function using {func}`~.lambdify`
+3. solve it numerically using SciPy's `solve_ivp`.
+
+```{warning}
+{func}`~.lambdify` uses {external:func}`~.exec` to dynamically execute Python code, and thus should not be used on unsanitized input.
+```
+
+Here is an [example from the field of chemical
+kinetics](https://www.sympy.org/scipy-2017-codegen-tutorial/notebooks/25-chemical-kinetics-intro.html):
+
+```py
+>>> from sympy import symbols, lambdify
+>>> import numpy as np
+>>> import scipy.integrate
+>>> def rhs(t, y, kf, kb):
+...     rf = kf * y[0]**2 * y[1]
+...     rb = kb * y[2]**2
+...     return [2*(rb - rf), rb - rf, 2*(rf - rb)]
+>>> y, (kf, kb) = symbols('y:3'), symbols('kf kb')
+>>> ydot = rhs(None, y, kf, kb)
```

Comment:
```suggestion
>>> ydot = rhs(None, y, kf, kb)
>>> ydot
```

Display ydot so that they see it is symbolic at this stage.

Reply:
Also, passing in `None` for `t` is probably not clear for a learner. Using the same `rhs` to both construct the symbolic form and evaluate the numerics is a bit of a trick that will be lost on learners. I think creating the symbolic form separately and explicitly, as you do in prior examples, is more clear for a learner. If you do that, then you can avoid these tricks for shortening the code.